src/shared/containers/ResourceViewContainer.tsx

```diff
@@ -0,0 +1,317 @@
+  const goToResource = (
+    orgLabel: string,
+    projectLabel: string,
+    resourceId: string,
+    opt: {
+      revision?: number;
+      tab?: string;
+      expanded?: boolean;
+    }
+  ) => {
+    const { revision, tab, expanded } = opt;
+    const pushRoute = `/${orgLabel}/${projectLabel}/resources/${resourceId}${
+      revision ? `?rev=${revision}` : ''
+    }${expanded ? '&expanded=true' : ''}${tab ? tab : ''}`;
+    history.push(pushRoute, location.state);
+  };
+  const goToProject = (orgLabel: string, projectLabel: string) =>
+    history.push(`/${orgLabel}/${projectLabel}`, location.state);
+  const goToOrg = (orgLabel: string) =>
+    history.push(`/${orgLabel}`, location.state);
+  const { expanded: expandedFromQuery, rev } = queryString.parse(
+    location.search
+  );
+  const activeTabKey = location.hash || DEFAULT_ACTIVE_TAB_KEY;
+
+  const [{ busy, resource, error }, setResource] = React.useState<{
+    busy: boolean;
+    resource: Resource | null;
+    error: Error | null;
+  }>({
+    busy: false,
+    resource: null,
+    error: null,
+  });
+  const [latestResource, setLatestResource] = React.useState<
+    (Resource & { [key: string]: any }) | null
+  >(null);
+
+  const isLatest =
+    (latestResource && latestResource._rev) === (resource && resource._rev);
+
+  const handleTabChange = (activeTabKey: string) => {
+    goToResource(orgLabel, projectLabel, resourceId, {
+      revision: resource ? resource._rev : undefined,
+      tab: activeTabKey,
+    });
+  };
+
+  const handleExpanded = (expanded: boolean) => {
+    goToResource(orgLabel, projectLabel, resourceId, {
+      expanded,
+      revision: resource ? resource._rev : undefined,
+      tab: activeTabKey,
+    });
+  };
+
+  const handleEditFormSubmit = async (value: any) => {
+    if (resource) {
+      try {
+        setResource({
+          resource,
+          error: null,
+          busy: true,
+        });
+        const { _rev } = await nexus.Resource.update(
+          orgLabel,
+          projectLabel,
+          resourceId,
+          resource._rev,
+          value
+        );
+        goToResource(orgLabel, projectLabel, resourceId, { revision: _rev });
+        notification.success({
+          message: 'Resource saved',
+          description: getResourceLabel(resource),
+          duration: 2,
+        });
+      } catch (error) {
+        notification.error({
+          message: 'An unknown error occurred',
+          description: error.message,
+          duration: 0,
+        });
+        setResource({
+          error,
+          resource: null,
+          busy: false,
+        });
+      }
+    }
+  };
+
+  const handleGoToInternalLink = (link: ResourceLink) => {
+    const { orgLabel, projectLabel } = getOrgAndProjectFromProjectId(
+      (link as IncomingLink)._project
+    );
+    goToResource(orgLabel, projectLabel, encodeURIComponent(link['@id']), {
+      tab: '#links',
+    });
+  };
+
+  useAsyncEffect(async () => {
```

Comment:
Can we use regular useEffect? In the spirit of removing useAsyncEffect from the code base?